### PR TITLE
Allow variant inventory quantity

### PR DESCRIPTION
**Pull request by mllemango**
As suggested in #801, Variant's`inventory_quantity` is now a read-only field, and should be readable from the object. 
The PR allows `inventory_quantity` to be filtered in as part of Variant's attributes. The field is only filtered out once variant is saved. User's will still be able to access the `inventory_quantity` field as an attribute of Variant and from the json object both before and after a save.

Also removed tests that are focused on deprecated versions, as well as the `allow_inventory_params?` method since that was also based on deprecated versions of the API.

### CODE REVIEW

File: lib/shopify_api/resources/variant.rb [hunk 1]
```diff
@@ -26,10 +26,9 @@ def old_inventory_quantity=(new_value)
       super
     end
 
-    def serializable_hash(options = {})
-      super(options).tap do |resource|
-        (resource['variant'] || resource).except!('inventory_quantity', 'old_inventory_quantity')
-      end
+    def save
+      attributes.except!('inventory_quantity', 'old_inventory_quantity')
+      super
```
Comment (paulomarg) [inline lib/shopify_api/resources/variant.rb]: This is great, though we are still dropping the fields. Would it make sense to restore the original attributes in an `ensure` block?

File: test/variant_test.rb [hunk 2]
```diff
@@ -35,30 +35,10 @@ def test_delete_variant
...
-    @variant.inventory_quantity_adjustment = 8
-  end
-
-  def test_setting_variant_inventory_quantity_adjustment_fails_in_2019_10_api
-    ShopifyAPI::Base.api_version = '2019-10'
-    refresh_variant(api_version: ShopifyAPI::Base.api_version)
-    assert_raises(ShopifyAPI::ValidationException) do
-      @variant.inventory_quantity_adjustment = 8
-    end
+    assert(10, @variant.inventory_quantity)
```
Comment (andyw8) [inline test/variant_test.rb:41]: Should be `assert_equal`? (below too)

Comment (mllemango) [reply test/variant_test.rb:41]: oops, yes it should be

File: lib/shopify_api/resources/variant.rb (hunk 1 above)
Comment (mllemango) [reply lib/shopify_api/resources/variant.rb]: Resolved offline, original attributes are still present after the save!

**Review by paulomarg — APPROVED** [review]

**Pull request merged**